### PR TITLE
Promote MetricProducer and MetricReader.RegisterProducer specification to feature-freeze

### PR DESCRIPTION
**Pull request by dashpole**
Part of #3599.  See the issue for the rationale for promotion.

### CODE REVIEW

**Comment by dashpole** [comment]
cc @open-telemetry/specs-metrics-approvers

**Comment by reyang** [comment]
@dashpole for the PR, I suggest we gather approvals from language SIG maintainers who have done prototyping.

**Review by reyang — APPROVED** [review]
Approve assuming we'll gather at least 3 approvals from language SIG maintainers who have done the prototype.

**Comment by dashpole** [comment]
@aabmass do you know which maintainers would be able to vouch for the JS implementation?
@MrAlias for go. @jack-berg for Java

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -1422,7 +1422,8 @@ A `MetricProducer` MUST support the following functions:
 `Produce` provides metrics from the MetricProducer to the caller. `Produce`
 MUST return a batch of [Metric points](./data-model.md#metric-points).
 `Produce` does not have any required parameters, however, [OpenTelemetry
-SDK](../overview.md#sdk) authors MAY choose to add parameters (e.g. timeout).
+SDK](../overview.md#sdk) authors MAY choose to add required or optional
+parameters (e.g. timeout).
```
Comment (damemi) [inline specification/metrics/sdk.md:1426]: sgtm for #3601

**Review by damemi — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Comment by dashpole** [comment]
I split out the parameter clarification into its own PR: https://github.com/open-telemetry/opentelemetry-specification/pull/3612.  This now only promotes MetricProducer

**Comment by dashpole** [comment]
Converting to draft while we discuss alternatives in https://github.com/open-telemetry/opentelemetry-specification/issues/3599.

**Pull request closed without merging**